### PR TITLE
Clean up underapproximate

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

File: src/Approximations/underapproximate.jl [hunk 1]
```diff
@@ -14,16 +14,27 @@ Compute the underapproximation of a convex set by sampling support vectors.
 
 The `VPolytope` obtained by taking the convex hull of the support vectors of `X`
 along the directions determined by `dirs`.
+
+### Notes
+
+Since the support vectors are not always unique, this algorithm may return
+a strict underapproximation even if the set can be exactly approximated using
+the given template. For example,
```
Comment (mforets) [inline src/Approximations/underapproximate.jl:22]: adding an example here may be relevant

File: src/Approximations/underapproximate.jl [hunk 2]
```diff
@@ -14,16 +14,27 @@ Compute the underapproximation of a convex set by sampling support vectors.
...
 """
 function underapproximate(X::LazySet{N}, dirs::AbstractDirections;
-                          apply_convex_hull::Bool=false) where {N<:Real}
-    @assert LazySets.dim(X) == LazySets.dim(dirs)
-    vinner = Vector{Vector{N}}(undef, length(dirs)) # TODO could be generalized to eltype(dirs)
+                          apply_convex_hull::Bool=false) where {N}
+    @assert dim(X) == dim(dirs) "the dimension of the set, $(dim(X)), doesn't match " *
+                                "the dimension of the template directions, $(dim(dirs))"
+    vinner = Vector{Vector{N}}(undef, length(dirs))
```
Comment (mforets) [inline src/Approximations/underapproximate.jl:28]: the problem of using `eltype(dirs)` is that such vector type is not always preserved. for instance if it's a SEV, the support vectors can be full. there are of course cases where keeping the vector type is relevant.

**Review by schillic — APPROVED** [review]
I suggest that we merge this.

**Comment by mforets** [comment]
ok let's merge when CI finishes

in the last commit i added an in-place version, with the purpose of generalizing the output array type

**Pull request merged**